platform/run_image.go

```diff
@@ -0,0 +1,78 @@
+package platform
+
+import (
+	"encoding/json"
+	"errors"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+const (
+	TargetLabel                = "io.buildpacks.target"
+	OSDistributionNameLabel    = "io.buildpacks.distribution.name"
+	OSDistributionVersionLabel = "io.buildpacks.distribution.version"
+)
+
+type TargetMetadata struct {
+	ID           string          `toml:"id"`
```

Comment:
heh -- i also just added this ID Field. but i think here you got rid of that "partial" struct i was playing with... pros and cons, but now it's decoupled from what happens in the buildpack side, which imo was the only value of that partial (to preserve & codify the coupling).  I thought it was correct to keep them coupled (and add the ID field to both structs) but if we're not doing that then it's probably correct to delete the TargetPartial struct entirely from bp_descriptor.go.

Reply:
I removed it just because it's simpler, but if we find we need it, we can put it back.